File: frontend/src/features/auth/services/authService.ts
import apiClient from '@/core/api/client';
import { ApiResponse } from '@/types/ApiResponse';
import { LoginResponse, UserDto } from '../types';

export const authService = {
  register: async (data: any): Promise<UserDto> => {
    const response = await apiClient.post<ApiResponse<UserDto>>('/auth/register', data);
    return response.data.data;
  },

  checkEmail: async (email: string): Promise<boolean> => {
    const response = await apiClient.post<ApiResponse<boolean>>('/auth/check-email', { email });
    return response.data.data;
  },

  verifyEmail: async (token: string): Promise<void> => {
    await apiClient.post<ApiResponse<void>>(`/auth/verify-email?token=${encodeURIComponent(token)}`);
  },

  resendVerification: async (email: string): Promise<void> => {
    await apiClient.post<ApiResponse<void>>('/auth/resend-verification', { email });
  },

  login: async (data: any): Promise<LoginResponse> => {
    const response = await apiClient.post<ApiResponse<LoginResponse>>('/auth/login', data);
    return response.data.data;
  },

  logout: async (refreshToken: string): Promise<void> => {
    await apiClient.post<ApiResponse<void>>('/auth/logout', { refreshToken });
  },

  forgotPassword: async (email: string): Promise<void> => {
    await apiClient.post<ApiResponse<void>>('/auth/forgot-password', { email });
  },

  resetPassword: async (data: any): Promise<void> => {
    await apiClient.post<ApiResponse<void>>('/auth/reset-password', data);
  }
};

File: backend/src/main/java/com/expenseflow/service/AuthService.java
package com.expenseflow.service;

import com.expenseflow.dto.auth.LoginRequest;
import com.expenseflow.dto.auth.LoginResponse;
import com.expenseflow.dto.auth.RegisterRequest;
import com.expenseflow.dto.auth.ResetPasswordRequest;
import com.expenseflow.dto.auth.LoginContext;
import com.expenseflow.dto.user.UserDto;

public interface AuthService {
    UserDto register(RegisterRequest request);
    LoginResponse login(LoginRequest request, LoginContext loginContext);
    void logout(String refreshToken);
    void verifyEmail(String token);
    void resendVerification(String email);
    void forgotPassword(String email);
    void resetPassword(ResetPasswordRequest request);
}
